### ISSUE TITLE
Integrate with buildkit

### ISSUE THREAD

**Issue by hinshun** [hinshun/ipcs#6]
POC `buildkit` using `ipcs` as psuedo-local build cache.

1. How do define a "cache-miss". For instance, LLB vertex is cacheable with hash `sha256:foobar`, it's impossible to know if the IPFS network does not have content for that hash. We either traverse the full network, limit hops to certain depth, set a timeout on fetching the content. On a separate topic, can I race a LLB vertex execution to fetching its cache?

**Comment by AkihiroSuda** [comment]
Execute op and attempt to pull the cache in parallel, and cancel the op if the cache is found?

**Comment by hinshun** [comment]
Yes exactly, thanks for commenting @AkihiroSuda. I only just started digging through buildkit's cache code, so I'm not that familiar with it yet. For buildkit's containerd worker, does buildkit cache using containerd's content store through its [proxy content store](https://github.com/containerd/containerd/blob/fe58b496bb7c4a40469f36499c2e837936abb7ae/content/proxy/content_store.go#L37)? If so, how does the runc worker cache?
